library/std/src/sys/sync/rwlock/solid.rs

```diff
@@ -83,6 +83,11 @@ impl RwLock {
         let rwl = self.raw();
         expect_success_aborting(unsafe { abi::rwl_unl_rwl(rwl) }, &"rwl_unl_rwl");
     }
+
+    #[inline]
+    pub unsafe fn downgrade(&self) {
+        todo!()
```

Comment:
The queue-based implementation [works in such a way](https://github.com/rust-lang/rust/issues/128203#issuecomment-2253247952) that `downgrade` can only be a hint, there cannot be a guarantee that readers actually get access. Therefore, and since unlocking is independent of the lock type on SOLID, this can just be a no-op.
```suggestion
        // Unlocking works the same way for read- and write-locks, so just leave the lock write-locked.
```

Reply:
I think that if the SOLID platform has a downgrade method on their locks that should still be preferred, but I agree with you if that is not the case.

However, I think that if it is possible for an implementation to allow waiting readers to gain access after a call to `downgrade`, that is still beneficial, even if the queue implementation does not. For example, in my proposed futex implementation it is true that not all waiting readers will get access, but at the very least some of them will win an [atomic update race](https://github.com/rust-lang/rust/pull/128219/files#diff-9a900676444ba4b5d6dc1375805a35295ad1e6fe97af126824711951536960d8R128) and get the read lock. 

If we don't allow waiting readers to read after a call to `downgrade`, then I feel the whole point of introducing `downgrade` is somewhat muted. Only the writer who had the write lock will get to read, which seems like the opposite of what someone calling `downgrade` to change the lock into shared mode would want.